daemon/api_test.go

```diff
@@ -1336,22 +1336,34 @@ func (s *apiSuite) TestConnectPlugSuccess(c *check.C) {
 		"status_code": 200.0,
 		"type":        "sync",
 	})
-	for slot, plugs := range d.interfaces.ConnectedSlots("consumer") {
-		c.Check(slot.Snap, check.Equals, "consumer")
-		c.Check(slot.Name, check.Equals, "slot")
-		for _, plug := range plugs {
-			c.Check(plug.Snap, check.Equals, "producer")
-			c.Check(plug.Name, check.Equals, "plug")
-		}
-	}
-	for plug, slots := range d.interfaces.ConnectedPlugs("producer") {
-		c.Check(plug.Snap, check.Equals, "producer")
-		c.Check(plug.Name, check.Equals, "plug")
-		for _, slot := range slots {
-			c.Check(slot.Snap, check.Equals, "consumer")
-			c.Check(slot.Name, check.Equals, "slot")
-		}
-	}
+	c.Assert(d.interfaces.Interfaces(), check.DeepEquals, &interfaces.Interfaces{
+		Plugs: []*interfaces.Plug{
+			&interfaces.Plug{
```

Comment:
Sorry, the proper question is actually this: the design doesn't give special importance to individual plugs and slots. They are already copies, which means there's little gain in having them as a pointer in a slice. Slices of pointers waste much more by having individual memory areas loose.

This is a bit of an early optimization, which is often ignored in the name of convenience or for the benefit of holding on to the value's _identity_ (its address). It doesn't feel like we benefit from either in this case.